pkg/specgen/generate/storage.go

```diff
@@ -81,14 +81,38 @@ func finalizeMounts(ctx context.Context, s *specgen.SpecGenerator, rt *libpod.Ru
 	}
 
 	for _, v := range s.Volumes {
-		if err = parse.ValidateVolumeCtrDir(v.Dest); err != nil {
-			return nil, nil, nil, err
+		// check if named volume has to be overlayed
+		overlayFlag := false
+		for _, o := range v.Options {
+			if o == "O" {
+				overlayFlag = true
+			}
 		}
-		cleanDestination := filepath.Clean(v.Dest)
-		if _, ok := unifiedVolumes[cleanDestination]; ok {
-			return nil, nil, nil, errors.Wrapf(errDuplicateDest, "conflict in specified volumes - multiple volumes at %q", cleanDestination)
+
+		if overlayFlag {
+			vol, err := rt.LookupVolume(v.Name)
+			if err != nil {
+				return nil, nil, nil, err
+			}
+			mountPoint, err := vol.MountPoint()
+			if err != nil {
+				return nil, nil, nil, err
+			}
+			newOverlayVol := new(specgen.OverlayVolume)
```

Comment:
I don't think this is viable. We now have a doubly-defined volume in the state, which seems dangerous - what if the named volume requires a mount, and the overlay volume is mounted before it, for example? Furthermore, you now have two separate mounts at the same mountpoint defined, which is going to mess up things like `podman inspect` and `podman run --volumes-from`.

Reply:
@mheon oh this happens when user specifically adds `:O` with the named volume for instance `podman container run -v example:/example:O busybox sh -c 'ls /example'` in all the cases `:O` is not configured for named volume it only happens when user explicitly specifies so in such cases named volume is only mounted as overlay inside container and not processed as a regular volume. So this keeps original named volume intact.

Reply:
@mheon so whenever it is mounted as overlay its not added to the list of `unifiedVolumes` but i can confirm if there are any race conditions.

Reply:
ps: we had a discussion and current blocker is `guarantee is removed` if user is using a overlayed named volume and lower is removed. cc @mheon

Reply:
this updated in latest commit.